config.go

```diff
@@ -12,168 +12,200 @@
+const (
+	FormatTypeRaw        = "raw"
+	FormatTypeStructured = "structured"
+	FormatTypeFile       = "file"
 )
 
-type RecordConfig struct {
-	FormatType    string
-	FormatOptions interface{}
-	Operation     string
+type Config struct {
+	Burst ConfigBurst `json:"burst"`
+	// Number of records to be generated (0 means infinite).
+	RecordCount int `json:"recordCount" validate:"gt=-1"`
+	// The time it takes to 'read' a record.
+	// Deprecated: use `rate` instead.
+	ReadTime time.Duration `json:"readTime"`
+	// The maximum rate in records per second, at which records are generated (0
+	// means no rate limit).
+	Rate float64 `json:"rate"`
+
+	// Configuration for default collection (i.e. records without a collection).
+	// Kept for backwards compatibility.
+	ConfigCollection
+	Collections map[string]ConfigCollection `json:"collections"`
 }
 
-func ParseRecordConfig(formatType, formatOptions, operation string) (RecordConfig, error) {
-	c := RecordConfig{
-		FormatOptions: make(map[string]interface{}),
-		Operation:     operation,
-	}
-	// check if it's a recognized format
-	switch formatType {
-	case FormatFile:
-		c.FormatType = FormatFile
-		if formatOptions == "" {
-			return RecordConfig{}, errors.New("file path not specified")
-		}
-		c.FormatOptions = formatOptions
-	case FormatStructured, FormatRaw:
-		fields, err := parseFields(formatOptions)
-		if err != nil {
-			return RecordConfig{}, fmt.Errorf("failed parsing fields: %w", err)
-		}
-		c.FormatType = formatType
-		c.FormatOptions = fields
-	default:
-		return RecordConfig{}, fmt.Errorf("unknown payload format %q", formatType)
-	}
+type ConfigBurst struct {
+	// The time the generator "sleeps" between bursts.
+	SleepTime time.Duration `json:"sleepTime"`
+	// The amount of time the generator is generating records in a burst. Has an
+	// effect only if `burst.sleepTime` is set.
+	GenerateTime time.Duration `json:"generateTime" default:"1s"`
+}
 
-	return c, nil
+type ConfigCollection struct {
+	// Comma separated list of record operations to generate. Allowed values are
+	// "create", "update", "delete", "snapshot".
+	Operation []string     `json:"operation" default:"create" validate:"required"`
+	Format    ConfigFormat `json:"format"`
 }
 
-type Config struct {
-	RecordCount  int64
-	ReadTime     time.Duration
-	SleepTime    time.Duration
-	GenerateTime time.Duration
-	RecordConfig RecordConfig
+type ConfigFormat struct {
+	// The format of the generated payload data (raw, structured, file).
+	Type string `json:"type" validate:"inclusion=raw|structured|file"`
+	// The options for the format type selected, which are:
+	//   1. For raw and structured: pairs of field names and field types, where the type can be one of: int, string, time, bool.
+	//   2. For the file format: a path to the file.
+	Options map[string]string `json:"options"`
+	// Path to the input file (only applicable if the format type is file).
+	FileOptionsPath string `json:"options.path"`
```

Comment:
Could we maybe remove this in the future if `Options` is now a type `map[string]string`. This means we could technically do `options.path` as well, right? And include that in the `Validate` method.

Reply:
We could do what you suggest. That said, I added this intentionally so that paramgen creates a parameter in the specs ([here](https://github.com/ConduitIO/conduit-connector-generator/blob/aec84121bf71fd7fb19d81fcffd06bd69ad9f740/config_paramgen.go#L58-L63)). This way it can be displayed in the UI and elsewhere, while if we remove it we have no idea that the connector can expect this parameter.

Reply:
@lovromazgon good point on the automatic generated documentation. Makes sense 👍